changes/2020-07-13_caching-cmm-max-plaintext-length/change.md

```diff
@@ -0,0 +1,153 @@
+
+### Fulfulling Encryption Materials Requests Without `max plaintext length`
+
+The Get Encryption Materials operation of the Cryptographic Materials Manager (CMM)
+accepts an Encryption Materials Request that MAY specify a `max plaintext length` field,
+indicating the maximum length of plaintext
+that the caller intends to encrypt using the returned materials.
+The caching Cryptographic Materials Manager (caching CMM)
+aims to prevent its users from using encryption materials
+to encrypt more bytes than its configured `byte limit`,
+and existing implementations do so by bypassing the cache
+whenever it does not know their users' `max plaintext length`.
+This behavior is safe but previously unspecified,
+and this change mandates that implementations implement it.
+
+### Setting `max plaintext length` for Underlying CMM's Get Encryption Materials
+
+The CMM interface specification implicitly allows an underlying CMM
+to produce encryption materials using logic
+that depends on the `max plaintext length` of its Encryption Materials Request.
+Suppose the caching CMM passes a `max plaintext length` value of `N`
+to its underlying CMM's Get Encryption Materials operation,
+and caches the materials that it receives.
+If a later Get Encryption Materials operation fetches those materials from the cache,
+but has a `max plaintext length` value that exceeds `N`,
+then the caching CMM returns those materials unsafely.
+
+It is therefore important that the caching CMM
+carefully selects the `max plaintext length` parameter to pass to its underlying CMM.
+This change mandates that the caching CMM passes its configured `byte limit` value
+as the `max plaintext length` parameter to the underlying CMM,
+whereas previously, the caching CMM could choose any value.
+With this change in place,
+even if the underlying CMM produces materials using plaintext-length-dependent logic,
+the caching CMM receives and caches materials
+that are safe to use for up to `byte limit` bytes of plaintext.
+It follows that if a later Get Encryption Materials operation's `max plaintext length`
+does not exceed the caching CMM's `byte limit`,
+then it is also safe for the caching CMM to return corresponding cached materials.
+
+## Drawbacks
+
+Bypassing the cache in the caching CMM on Get Encryption Materials operations
+that do not specify `max plaintext length`
+could be confusing to the user.
+Warning the user when this happens could help prevent users from running into
+unexpected (and potentially expensive) calls to the underlying CMM,
+but we leave this as an implementation choice.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any customer use case
+that depends on the Caching CMM passing no `max plaintext length` parameter
+to the underlying CMM on Get Encryption Materials calls.
+
+## Guide-level Explanation
+
+When you make a Get Encryption Materials call to the caching CMM,
```

Comment:
here and throughout, I think we should frame this in terms of "a caller" instead of "you". A caller could be the encrypt operation, or it could be another CMM.